### PR TITLE
global efficency

### PR DESCRIPTION
**Pull request by egolol**
I'm thinking of putting the ensembles of each here

### CODE REVIEW

File: braph2genesis/measures/_GlobalEfficiency.gen.m [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    ];
+
+known_global_efficiency = {[1/4 1/4 1/4 1/4 0]'};
+
+g = GraphBU('B', B);
+global_efficiency = GlobalEfficiency('G', g).get('M');
+
+assert(isequal(global_efficiency, known_global_efficiency), ...
+    [BRAPH2.STR ':GlobalEfficiency:' BRAPH2.BUG_ERR], ...
+    'GlobalEfficiency is not being calculated correctly for GraphBU.')
```
Comment (annacanal) [inline braph2genesis/measures/_GlobalEfficiency.gen.m:64]: GraphWU

File: braph2genesis/measures/_GlobalEfficiency.gen.m [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    0   .1  0   0   0
+    .2  0   0   0   0
+    0   0   0   .2  0
+    0   0   .1  0   0
+    0   0   0   0   0
+    ];
+
+known_global_efficiency = {[1/4 1/4 1/4 1/4 0]'};
+
+g = GraphBU('B', B);
```
Comment (annacanal) [inline braph2genesis/measures/_GlobalEfficiency.gen.m:59]: this should be GraphWU

File: braph2genesis/measures/_GlobalEfficiency.gen.m [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      ];
+B = {B11  B22};
+
+known_global_efficiency = {
+                        [1/4 1/4 1/4 1/4 0]'
+                        [1/4 1/4 1/4 1/4 0]'
+                        };
+
+
+g = MultiplexGraphBU('B', B);
```
Comment (annacanal) [inline braph2genesis/measures/_GlobalEfficiency.gen.m:93]: MultiplexGraphWU or MultiplexGraphBU?

File: braph2genesis/measures/_GlobalEfficiency.gen.m [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+global_efficiency = GlobalEfficiency('G', g).get('M');
+
+assert(isequal(global_efficiency, known_global_efficiency), ...
+    [BRAPH2.STR ':GlobalEfficiency:' BRAPH2.BUG_ERR], ...
+    'GlobalEfficiency is not being calculated correctly for MultiplexGraphBU.')
```
Comment (annacanal) [inline braph2genesis/measures/_GlobalEfficiency.gen.m:98]: MultiplexGraphWU or MultiplexGraphBU?

File: braph2genesis/measures/_GlobalEfficiencyAv.gen.m [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    global_efficiency_av(li) = {mean(global_efficiency{li})};
+end
+
+value = global_efficiency_av;
+
+%% ¡tests!
+
+%%% ¡test!
+%%%% ¡name!
+GraphWU
```
Comment (annacanal) [inline braph2genesis/measures/_GlobalEfficiencyAv.gen.m:42]: the title doesnt match the test GraphBU
You could also add at list one test for weighted graphs (from the old measure in braph2)

File: braph2genesis/measures/_InGlobalEfficiency.gen.m [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    0   .1  0   0   0
+    .2  0   0   0   0
+    0   0   0   .2  0
+    0   0   .1  0   0
+    0   0   0   0   0
+    ];
+
+known_in_global_efficiency = {[1/4 1/4 1/4 1/4 0]'};
+
+g = GraphBD('B', B);
```
Comment (annacanal) [inline braph2genesis/measures/_InGlobalEfficiency.gen.m:59]: same than before.. you just the binary tests. but have Weighted in title

File: braph2genesis/measures/_OutGlobalEfficiency.gen.m [hunk 7]
```diff
@@ -0,0 +1,98 @@
+%% ¡header!
+OutGlobalEfficiency < Measure (m, out-global efficiency) is the graph out-global efficiency.
+
+%%% ¡description!
+The out-global efficiency is the average outverse shortest out-path length without each layer. 
+It is outversely related to the characteristic out-path length.
```
Comment (annacanal) [inline braph2genesis/measures/_OutGlobalEfficiency.gen.m:6]: inversely ?

File: braph2genesis/measures/_OutGlobalEfficiency.gen.m [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    global_efficiency_layer = (sum(outverse_distance, 2) / (N(li)-1));
+    out_global_efficiency(li) = {global_efficiency_layer};
+end
+value = out_global_efficiency;
+
+%% ¡tests!
+
+%%% ¡test!
+%%%% ¡name!
+GraphWD
```
Comment (annacanal) [inline braph2genesis/measures/_OutGlobalEfficiency.gen.m:47]: same

**Review by annacanal — CHANGES_REQUESTED** [review]
small details

**Review by c-yuwei — COMMENTED** [review]
reviewed along with @annacanal 's comments

**Review by annacanal — APPROVED** [review]

**Review by giovannivolpe — APPROVED** [review]

**Pull request merged**